tests/rptest/clients/kubectl.py

```diff
@@ -133,14 +134,25 @@ def _install(self):
             self._kubectl_installed = True
         return
 
+    @property
+    def logger(self) -> Logger:
+        return self._redpanda.logger
+
     def _cmd(self, cmd):
         # Log and run
         ssh_prefix = self._ssh_prefix()
         remote_cmd = ssh_prefix + cmd
         self._redpanda.logger.info(remote_cmd)
-        return subprocess.check_output(remote_cmd)
+        try:
+            return subprocess.check_output(remote_cmd, stderr=subprocess.PIPE)
```

Comment:
The issue was two-fold: we don't capture `stderr` and when an excpetion occurs we don't log anything and the exception text only logs the command and the failure code, not the output.